### PR TITLE
ci: enable parallelism for all integration jobs

### PR DESCRIPTION
**Pull request by majorgreys**
## Description

Use CircleCI parallelism for all integrations jobs (with the exception of `mysqldb` since it has fewer than 4 environments to run). 

## Checklist
- [ ] Entry added to release notes using [reno](https://docs.openstack.org/reno/latest/user/usage.html)
- [ ] Tests provided; and/or
- [ ] Description of manual testing performed and explanation is included in the code and/or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -134,7 +138,7 @@ commands:
         type: string
       split:
```
Comment (Kyle-Verhoog) [inline .circleci/config.yml:139]: can we do away with split (and the environment variable)?

Comment (majorgreys) [reply .circleci/config.yml]: done!

File: .circleci/config.yml [hunk 2]
```diff
@@ -125,16 +129,13 @@ commands:
                 command: pip install tox && tox -e 'wait' << parameters.wait >>
       - run:
           name: "Running tests matching '<< parameters.pattern >>'"
-          command: "riot -v run -s '<< parameters.pattern >>'"
+          command: "echo 2.7,3.5,3.6,3.7,3.8,3.9 | tr ',' '\n' | circleci tests split | xargs -I PY riot -v run -pPY -s '<< parameters.pattern >>'"
```
Comment (Kyle-Verhoog) [inline .circleci/config.yml:132]:
```suggestion
          command: "echo -p2.7,-p3.5,-p3.6,-p3.7,-p3.8,-p3.9 | tr ',' '\n' | circleci tests split | xargs riot -v run -s '<< parameters.pattern >>'"
```

**Review by Kyle-Verhoog — APPROVED** [review]
This has brought our CI time down to below [20 minutes](https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/3524/workflows/9b1e331b-fd41-417d-b89c-a0831789d5e4)!!! 🚀 🚀 🚀

File: .circleci/config.yml [hunk 3]
```diff
@@ -125,16 +129,13 @@ commands:
                 command: pip install tox && tox -e 'wait' << parameters.wait >>
       - run:
           name: "Running tests matching '<< parameters.pattern >>'"
-          command: "riot -v run -s '<< parameters.pattern >>'"
+          command: "echo -p2.7,-p3.5,-p3.6,-p3.7,-p3.8,-p3.9 | tr ',' '\n' | circleci tests split | xargs riot -v run -s '<< parameters.pattern >>'"
```
Comment (brettlangdon) [inline .circleci/config.yml:132]: same comment as the other PR, I wonder if we should just build `CIRCLECI_NODE_INDEX` and `CIRCLECI_NODE_TOTAL` into `riot`, could make this much easier to manage.

Comment (majorgreys) [reply .circleci/config.yml:132]: Agree this is best done within riot!

**Pull request merged**